src/main/java/org/candlepin/subscriptions/security/IdentityHeaderAuthenticationManager.java

```diff
@@ -33,91 +31,47 @@
 import org.springframework.security.web.authentication.preauth.PreAuthenticatedGrantedAuthoritiesWebAuthenticationDetails;
 import org.springframework.util.StringUtils;
 
-import java.io.IOException;
-import java.util.Collections;
-import java.util.Map;
 
 /**
- * This class is in charge of deserializing the bytes from the x-rh-identity header and extracting the
- * account number from them.
+ * This class is responsible for validating the principal. If a valid principal was
+ * found, the request is considered authenticated.
  */
 public class IdentityHeaderAuthenticationManager implements AuthenticationManager {
 
-    private final ObjectMapper mapper;
-
-    public IdentityHeaderAuthenticationManager(ObjectMapper mapper) {
-        this.mapper = mapper;
-    }
-
     /**
-     * This method is a little odd.  Since the value of the x-rh-identity header is JSON, there is the
-     * potential for a JSON parsing error.  However, if we parse the JSON in
-     * {@link IdentityHeaderAuthenticationFilter}, when we throw an exception on a parse error, it is not
-     * handled very nicely.  By handling the parsing in an AuthenticationManager we follow the conventions
```

Comment:
Is this no longer the case since we return an empty principal now?

Reply:
Yes, this shouldn't be applicable anymore. Things seemed to behave correctly in the cases that I've tested. Please feel free to see if you can break it as in the original security PR.